### ISSUE TITLE
How do I include my snippets into the export?

### ISSUE THREAD

**Issue by pattystefy** [KosmosisDire/obsidian-webpage-export#361]
**This is more of a question than a bug**, but how do I include my snippets? I'm not familiar with CSS or snippets, I just like to change the appearance of my notes. I tried to add "custom head content" but no matter the extension, it doesn't recognize the path. It always says "Path must be: .html, .htm, .txt" but it's still not recognized after changing from the .css extension. 
I use Windows 10 Pro, and I have everything up-to-date. I don't think is a bug, at least I hope so.

**Comment by KosmosisDire** [comment]
The path must be .html thing is a bug. However, that is not how you add snippets. Snippets are a built in feature of obsidian, you can Google it, but they are accessible in obsidian's Appearance settings.

**Comment by pattystefy** [comment]
> The path must be .html thing is a bug. However, that is not how you add snippets. Snippets are a built in feature of obsidian, you can Google it, but they are accessible in obsidian's Appearance settings.

I know how to add them. What I meant was that I have some snippets that change my notes, like the card snippet or image adjustments. The first one transforms the lists into cards and the second one allows the text to wrap around the images when you move them to either right or left. I would love to have those changes in the notes when I export them too, but I'm not sure which setting I should modify for that.

**Comment by KosmosisDire** [comment]
Snippets are automatically included, you don't have to do anything. Is it not being included for you?

**Comment by pattystefy** [comment]
> Snippets are automatically included, you don't have to do anything. Is it not being included for you?

No, they aren't. I tried to update everything I had and I exported three times to make sure, but they aren't included.

**Comment by pattystefy** [comment]
I restarted it, but those two snippets still don't appear in the exported files... I did export again.

**Comment by KosmosisDire** [comment]
Can you run an export and then copy the debug info from the plugin's settings and paste it here? Then can you also go to your export, open the `/lib/styles/snippets.css` file (if it exists) and let me know if the snippet's text is in the file.

**Comment by pattystefy** [comment]
Sure! Here is the debug info:
[debug info.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14363963/debug.info.txt)

I put it in this txt file, I hope that's alright. And also, there isn't any "styles/snippets.css" in the "/lib".

**Comment by KosmosisDire** [comment]
Thanks, could you test an export using Online Web Server mode? I think it might work using that mode. If it does work in that mode then I think I know enough to be able to fix this.

**Comment by pattystefy** [comment]
Hello, I'm very sorry for the late update on the situation. I exported the notes, as you said, and the cards still don't work, but it looks like the other CSS (image adjustments) works. Also, the directory "styles/snippets.css" appeared this time. Here I copied the new debug info, I hope it's helpful:

[debug info.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14418825/debug.info.txt)

**Comment by KosmosisDire** [comment]
Thanks, no worries! If you open the `snippets.css` does it have all the css that you would expect it to have in it? Like all the cards snippets, etc? (as a side note, be aware that if obsidian updates to 1.5.8 it will break other styles. I am working on that right now)

**Comment by pattystefy** [comment]
Yes, it has the snippets I need, but I also have the updated version of Obsidian. I will wait for the update then, if you think that the version of the app is the issue. Thank you for the  help!

**Comment by KosmosisDire** [comment]
It's possible the obsidian version is the issue, yes. If it still persists after the fix I will look into it more. I will also make sure the snippets work on self-contained exports.

**Comment by KosmosisDire** [comment]
Can you please test out the new beta 1.8.1-0b and let me know if this fixes the issue? Please also let me know if you find any new issues. Thanks!

**Comment by pattystefy** [comment]
Hello! Thank you for the update. I tried to export using all 3 mods (Raw HTML Documents, HTML Documents, Online Web Server) in case at least one of them worked, and this is the result I got:

<img width="350" alt="Screenshot_5" src="https://github.com/KosmosisDire/obsidian-webpage-export/assets/149337812/e2cf640b-d5a9-41f7-a3f9-1f0b0aed6480">
<img width="329" alt="Screenshot_6" src="https://github.com/KosmosisDire/obsidian-webpage-export/assets/149337812/9c657e2d-a29f-43c6-a82f-091c30291411">
<img width="350" alt="Screenshot_8" src="https://github.com/KosmosisDire/obsidian-webpage-export/assets/149337812/fdac1710-6a21-4155-a3ff-8533935e3222">

And this is how it should have looked like:
<img width="400" alt="Screenshot_4" src="https://github.com/KosmosisDire/obsidian-webpage-export/assets/149337812/a82743bc-0d34-402a-9b99-fb5696f9e13e">

I am referring to the cards, of course. Also, I got this error:

<img width="417" alt="Screenshot_7" src="https://github.com/KosmosisDire/obsidian-webpage-export/assets/149337812/6e5fa9b2-0def-4e23-a443-d5de6db3b06d"> But I selected the folder, so it exists.

Last time I exported using Online Web Server, so this is the debug info for that:
[Debug Info.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14471022/Debug.Info.txt)

Then I exported again with HTML Documents, to get another debug info:
[Debug Info.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14471024/Debug.Info.txt)

I am sorry to give you bad news ._. I hope you'll be able to solve this.

**Comment by KosmosisDire** [comment]
No worries, is it possible for you to provide me with the snippets you are using?

**Comment by pattystefy** [comment]
Sure! These are the ones that I use:

[S - Images Adjustments.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14472708/S.-.Images.Adjustments.txt)
[cards.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14472709/cards.txt) This one, the cards have a problem.

And those are from my theme, Anupuccin, and I had no issues with them:
[floating-search-bar.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14472711/floating-search-bar.txt)
[custom-rainbow-colors.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14472712/custom-rainbow-colors.txt)
[extended-colorschemes.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14472713/extended-colorschemes.txt)

And this one is from the same theme, but I don't mind not appearing:
[custom-background.txt](https://github.com/KosmosisDire/obsidian-webpage-export/files/14472715/custom-background.txt)

**Comment by KosmosisDire** [comment]
So I just downloaded the cards snippet and enabled it in my test vault and it worked right away. Is it possible for you to zip and send me the whole html export so I can check why it isn't applying the styles? (Also if you can send me a .zip of a vault that reproduces the problem this would also be helpful) Thanks! :)

**Comment by Chesher** [comment]
Hi!
I'm having a similar issue - I'm super new to all this so this could be just me, but here is my CSS from my snippets folder:

.inline-title {
  color: #ffc106
}
.cm-header-1 { color: #ffc106; }  
.cm-header-2 { color: #ffc106; }
.cm-header-3 { color: #ffc106; }

It's just a super basic one allowing me to have some custom colors on Titles and Headers.
It looks as expected in Obsidian, but the colors don't show up in the exported HTML file.

Erm since this is a recolor of the default theme I'm really open to any other solutions to just have colored titles and headers in the exported html

**Comment by KosmosisDire** [comment]
> Hi!
> I'm having a similar issue - I'm super new to all this so this could be just me, but here is my CSS from my snippets folder:
> 
> .inline-title {
>   color: #ffc106
> }
> .cm-header-1 { color: #ffc106; }  
> .cm-header-2 { color: #ffc106; }
> .cm-header-3 { color: #ffc106; }
> 
> It's just a super basic one allowing me to have some custom colors on Titles and Headers.
> It looks as expected in Obsidian, but the colors don't show up in the exported HTML file.
> 
> Erm since this is a recolor of the default theme I'm really open to any other solutions to just have colored titles and headers in the exported html

So in your case, I believe this is because your styles only apply to the editing view not in reading mode. If you switch to reading mode in obsidian your styles won't show up even in obsidian.

**Comment by Chesher** [comment]
> > Hi!
> > I'm having a similar issue - I'm super new to all this so this could be just me, but here is my CSS from my snippets folder:
> > .inline-title {
> > color: #ffc106
> > }
> > .cm-header-1 { color: #ffc106; }
> > .cm-header-2 { color: #ffc106; }
> > .cm-header-3 { color: #ffc106; }
> > It's just a super basic one allowing me to have some custom colors on Titles and Headers.
> > It looks as expected in Obsidian, but the colors don't show up in the exported HTML file.
> > Erm since this is a recolor of the default theme I'm really open to any other solutions to just have colored titles and headers in the exported html
> 
> So in your case, I believe this is because your styles only apply to the editing view not in reading mode. If you switch to reading mode in obsidian your styles won't show up even in obsidian.

I knew it was something stupid like that on my end! Sorry for any troubles caused!

**Comment by KosmosisDire** [comment]
No worries! :)

**Comment by cabogie** [comment]
For anyone with a split config step like me - `.obsidandesktop` and `.obsidianmobile`

I was able to get snippets to work by copying them into an `.obsidian/snippets` folder. 

Perhaps this is something to add support for, but not a huge issue. In fact it's kinda nice, I'll probably make myself some different export styling (unless there's a better way to do that).

**Comment by KosmosisDire** [comment]
> I'll probably make myself some different export styling (unless there's a better way to do that).

Depending on the version you're using you can create export specific styles by prefixing any selector with 'publish'.

For example:
```css
publish h1
{
    color: red;
}
```

will only apply when exported. You can also include css using the custom head content if you'd like.

> For anyone with a split config step like me - .obsidandesktop and .obsidianmobile
> I was able to get snippets to work by copying them into an .obsidian/snippets folder.

I am not aware of how split configs work, I'll have to look into it to make sure they work. Is this something you set up manually or is this something obsidian sets up? How did you configure this?

**Comment by cabogie** [comment]
> Depending on the version you're using you can create export specific styles by prefixing any selector with 'publish'.
> 
> For example:
> 
> ```css
> publish h1
> {
>     color: red;
> }
> ```
> 
> will only apply when exported. You can also include css using the custom head content if you'd like.

Cool, thanks!

> I am not aware of how split configs work, I'll have to look into it to make sure they work. Is this something you set up manually or is this something obsidian sets up? How did you configure this?

It's under `Settings` > `Files and links` > `Override config folder` at the bottom.

Probably not a common use case tbf. I just personally like to keep my mobile and desktop configs separate and happen to not use `.obsidian` anymore.

**Comment by KosmosisDire** [comment]
I am aware of this option, but I thought that would only change the name of the folder, not allow you to have 2 separate configs. But either way the plugin should use the config folder that is currently set up, so I'll have to look into that to make sure it's still working.

**Comment by cabogie** [comment]
I think it's as simple as Obsidian just looks at that folder for config. So on my desktop / laptop it looks at the stuff in `.obsidiandesktop` and on my phone it looks at `.obsidianmobile`.

As a quick test I changed it to something new and it created a blank slate folder with the given name.

**Comment by d4vidyo** [comment]
Im trying to use the [Banner snippet](https://github.com/HandaArchitect/obsidian-banner-snippet) but it wont export correctly.
Here is how it looks in Obsidian:
![2024-12-09 21_25_32-NVIDIA GeForce Overlay DT](https://github.com/user-attachments/assets/ce3032c6-b50a-4045-a786-b95c0e17539c)

And here is how it looks after exporting it:
![2024-12-09 21_26_43-NVIDIA GeForce Overlay DT](https://github.com/user-attachments/assets/64281919-50de-4c85-b8ac-4e7fb2602594)

Im not sure what the issue is, i thought custom css should be applied automatically.

**Comment by KosmosisDire** [comment]
This banner snippet relies on specific elements and classes outside of just the document. Therefore it cannot be applied properly. Try using the banners plugin, it does the same thing and is supported by this plugin.

**Comment by AlexBoer** [comment]
This may be related. It seems that The Obsidian Webpage Export does not render the Minimal Theme "Cards" CSS class correctly. Not sure if this is a known issue or just something on my end though.

**Comment by AlexBoer** [comment]
[Minimal Theme Cards Not Working Web Files.zip](https://github.com/user-attachments/files/18941161/Minimal.Theme.Cards.Not.Working.Web.Files.zip)
Here's a copy of the export.

Here's a screenshot of what I expect to see: 
![Image](https://github.com/user-attachments/assets/fc3809e5-c8b1-4f52-b099-460201d9664b)

And heres a screenshot of what I see in the HTML files (both HTML documents and the online web export)
![Image](https://github.com/user-attachments/assets/5483d0fb-a1f9-40cf-bfb6-3b8f19ae093a)

**Comment by KosmosisDire** [comment]
@AlexBoer are you using the beta version? I am using minimal cards in testing and it's working for me

**Comment by AlexBoer** [comment]
> [@AlexBoer](https://github.com/AlexBoer) are you using the beta version? I am using minimal cards in testing and it's working for me

It does indeed work in the beta version! thank you!

**Comment by Nudnuy00** [comment]
CSS snippets are always treated with low priority during export. Currently, the only way to make them work is by adding `!important` to every single property, which is not ideal. Is it possible to load the snippets at the very end of the document to ensure they override the default styles?
